src/Server.zig

```diff
@@ -237,72 +237,82 @@ fn publishDiagnostics(server: *Server, writer: anytype, handle: DocumentStore.Ha
         });
     }
 
-    if (server.config.enable_unused_variable_warnings) {
-        scopes: for (handle.document_scope.scopes.items) |scope| {
-            const scope_data = switch (scope.data) {
-                .function => |f| b: {
-                    if (!ast.fnProtoHasBody(tree, f).?) continue :scopes;
-                    break :b f;
-                },
-                .block => |b| b,
-                else => continue,
+    if (server.config.enable_ast_check_diagnostics) diag: {
+        if (server.config.zig_exe_path) |zig_exe_path| {
+            var process = std.ChildProcess.init(&[_][]const u8{ zig_exe_path, "ast-check", "--color", "off" }, server.allocator);
+            process.stdin_behavior = .Pipe;
+            process.stderr_behavior = .Pipe;
+
+            process.spawn() catch |err| {
+                Logger.warn(server, writer, "Failed to spawn zig fmt process, error: {}", .{err});
```

Comment:
Won't `exec` block before I can write to its stdin?

Reply:
Sorry, i overlooked that you wrote the file to stdin. I would have given the file path as an argument to avoid writing the whole file to stdin. You can just keep it as it is if you want.

Reply:
No worries Techatrix! I think I'll leave it as-is because diagnostics don't always run on saved files that I can feed to ast-check which is why I use stdin.